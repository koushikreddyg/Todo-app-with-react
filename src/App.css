
.login-form {
  padding: 5em;
  text-align: center;
  background-color: #0099ff;
}

.login-form-button  {
  background: #0099ff;
  background-image: -webkit-linear-gradient(top, #0099ff, #000000);
  background-image: -moz-linear-gradient(top, #0099ff, #000000);
  background-image: -ms-linear-gradient(top, #0099ff, #000000);
  background-image: -o-linear-gradient(top, #0099ff, #000000);
  background-image: linear-gradient(to bottom, #0099ff, #000000);
  -webkit-border-radius: 8;
  -moz-border-radius: 8;
  border-radius: 8px;
  font-family: Georgia;
  color: #ffffff;
  font-size: 15px;
  padding: 9px 20px 10px 20px;
  text-decoration: none;
  margin-left: 5em;
  margin-top: 2em;
}
.login-form-button :hover {
  background: #ff0000;
  background-image: -webkit-linear-gradient(top, #ff0000, #000000);
  background-image: -moz-linear-gradient(top, #ff0000, #000000);
  background-image: -ms-linear-gradient(top, #ff0000, #000000);
  background-image: -o-linear-gradient(top, #ff0000, #000000);
  background-image: linear-gradient(to bottom, #ff0000, #000000);
  text-decoration: none; 
}
.todo{
  
text-align: center;
border-top: 5em;
}
.todo1{
  text-align: center; 
}
.centre{
  margin-left: 30em; 
}





