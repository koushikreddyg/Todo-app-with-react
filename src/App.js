import React, {Component} from 'react';
import Demoform from './demoform';
import {  Icon, Input, Button } from 'antd';
import './App.css';
class App extends Component{
  constructor(props) {
    super(props);
    this.state=({
      list:[],
      shown:true
    });
    this.todolist = this.todolist.bind(this);
    this.dele = this.dele.bind(this);
    this.edit = this.edit.bind(this);
    this.changevalue=this.changevalue.bind(this);

  }
  todolist(p) {
    let temp = this.state.list;
    temp.push(Object.assign({}, p));
    this.setState({
      list: temp
    });
  }
  dele(ind) {
    let temp = this.state.list;
    temp.splice(ind,1);
    this.setState({
      list: temp
    });
   
  }
  edit() {
    this.setState({
      shown:!this.state.shown
    });
  }
 changevalue(e,x,y){
var state=this.state;
state.list[x][y]=e.target.value;
this.setState(state);
 }
  render(){
    var show={
        display:this.state.shown? "block":"none"
    }
    var hide={
      display:this.state.shown? "none":"block"
  }
    return (
      <div className="todo">
        
      <div>
          <Demoform addlist={this.todolist}></Demoform>
          <div>
      <h1>TODO_List</h1>
      {
      this.state.list.map((i, ind) => {
        return (
          <span key={'key' + ind}>
            <b>Your Details {ind}:</b><br />
            {Object.keys(i).map((k, ind1) => {
              return (
                <span key={'send key' + ind1}>
                  {k}:<Input  prefix={<Icon type="user" style={{ color: 'rgba(0,0,0,.25)' }} />} type="text" style={hide} onChange={(e)=>this.changevalue(e,ind,k)} value={i[k]}/>
                  <p style={show}>{i[k]}</p> <br /></span>
              )
            })}
            
            <Button className="login-form-button" onClick={() => { this.edit() }}>edit</Button>
            
            <Button className="login-form-button" onClick={() => { this.dele(ind) }}>Delete</Button><br />
          </span>
        )
      })
    }
      </div>
      </div>
        </div>
    )
  };
}
export default App;