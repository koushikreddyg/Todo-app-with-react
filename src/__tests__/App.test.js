import React from 'react';
import toJSON from 'enzyme-to-json';
import {shallow} from 'enzyme'
import App from '../App';

test('test App.js File ',()=>{
   const wrapper=shallow(<App/>);
   expect(wrapper).toMatchSnapshot();
})