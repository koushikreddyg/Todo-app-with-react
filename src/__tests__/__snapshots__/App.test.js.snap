// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test App.js File  1`] = `
<div>
  <AppRouter />
</div>
`;
